tests/Aspire.Hosting.Tests/ProjectResourceTests.cs

```diff
@@ -591,6 +591,9 @@ public async Task AddProjectWithArgs()
         Assert.Collection(args,
             arg => Assert.Equal("arg1", arg),
             arg => Assert.Equal("http://localhost:1234", arg));
+
+        // We don't yet process relationships set via the callbacks
+        Assert.False(project.Resource.TryGetAnnotationsOfType<ResourceRelationshipAnnotation>(out var relationships));
```

Comment:
This is a case we're not handling https://github.com/dotnet/aspire/issues/8426

Reply:
How might we solve this in the future. I think it makes sense to immediately add resource relationships when you can like this, but being able to catch these late discovered relationships is also useful.

So do we put the logic in two places.

If we try to put the logic in one place like in DCP integration, then any non-DCP resources won't be able to play (anything that evaluates expressions would need to the logic to add these relationships).

Reply:
I stated that way but decided to back out for 9.2. We’d do it in the application orchestrator before start.